### ISSUE TITLE
rancher2_catalog_v2 not working behind proxy

### ISSUE THREAD

**Issue by DaniloE** [rancher/terraform-provider-rancher2#536]
Hi,

we want to use rancher2_catalog_v2 for adding the bitnami-Repository. We have to use a proxa and configured it via ENV-variables inside the Pod.
In V1 (rancher2_catalog) everything works fine, but with V2 we get

**Error: [ERROR] waiting for catalog (bitnami) to be active: Get "https://charts.bitnami.com/bitnami/index.yaml": dial tcp 99.86.2.104:443: connect: connection refused**

in terraform. 

As i said, no problem with V1, only V2, so it seems that there must be something inside the code that prevents using proxy-settings. Unfortunately my go is not good enough to find the spot.

We use:
```bash
Terraform v0.13.2
+ provider registry.terraform.io/hashicorp/dns v3.0.0
+ provider registry.terraform.io/hashicorp/helm v1.3.2
+ provider registry.terraform.io/hashicorp/kubernetes v1.13.3
+ provider registry.terraform.io/hashicorp/local v2.0.0
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.0
+ provider registry.terraform.io/hashicorp/tls v3.0.0
+ provider registry.terraform.io/rancher/rancher2 v1.10.6
+ provider registry.terraform.io/rancher/rke v1.1.4
```

Our definition in terraform:

```terraform
# resource rancher2_catalog bitnami {
#   for_each = rancher2_cluster.rke_cluster_app

#   cluster_id = each.value.id
#   name = "bitnami"
#   url = "https://charts.bitnami.com/bitnami"

#   version = "helm_v3"
# }

resource rancher2_catalog_v2 bitnami {
 for_each = rancher2_cluster.rke_cluster_app

 cluster_id = each.value.id
 name = "bitnami"
 url = "https://charts.bitnami.com/bitnami"
}
```

Thanks in advance,
Danilo

**Comment by DaniloE** [comment]
Great...after 5 hours i found a solution. We have to manually set the env-variables in the deployment-defintion of the cattle-agent pod.

```yaml
  env:
        - name: HTTP_PROXY
          value: http://$PROXY
        - name: HTTPS_PROXY
          value: http://$PROXY
        - name: NO_PROXY
          value: $NO_PROXY
        - name: http_proxy
          value: $PROXY
        - name: https_proxy
          value: $PROXY
```

It is a solution, but nor really practical at the moment, because we have several clusters.
The docker-daemon is via ENV configured to use the proxy, 

Did i forget something, or has anyone a better solution than manual change?

**Comment by eest** [comment]
While I dont know of a solution I think this is related to these tickets:
https://github.com/rancher/rancher/issues/24876
https://github.com/rancher/rancher/issues/29961
https://github.com/rancher/rancher/issues/30846

**Comment by rawmind0** [comment]
@eest , you are right, this issue is related to these Rancher issues. Fix is included on Rancher v2.5.6, nothing can be done from this tf provider.

Please, reopen the issue if needed
